### PR TITLE
Listing Detail Redesign (DApp3)

### LINKED ISSUE #2413: Improved Listing Detail Screen
Implement the redesigned listing detail screen as shown here:

**_Mobile_**
https://zpl.io/V4e3LOM
**_Desktop_**
https://zpl.io/b649zDR

We will attempt to mock up the various states of the "buy box", as it was previously known. But I'm not sure how soon this will happen.
- perspective: buyer, seller, neither
- listing types: unit, fractional, etc
- availability: quantity, pending, sold, etc

### PR DESCRIPTION
**Pull request by shahthepro**
Should close #2413 

### Checklist:
- [x] Check layout and various on mobile and desktop
  - [x] Single Unit
  - [x] Multi Unit
  - [x] Fractional/Nightly
  - [x] Fractional Hourly
- [x] Push to navigation stack wherever possible
- [x] Redesign Calendar Component
- [x] Test your work and double-check to confirm that you didn't break anything
- [x] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [x] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [x] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

File: dapps/marketplace/src/pages/listing/_ListingPending.js [hunk 1]
```diff
@@ -2,20 +2,32 @@ import React from 'react'
...
+      <div className="status-title">
+        <fbt desc="Pending">Pending</fbt>
       </div>
-    </fbt>
-    <Link to="/listings">
-      <fbt desc="viewListings">View Listings</fbt>
-    </Link>
+      <div className="status-text">
+        {!isMobile
+          ? fbt(
```
Comment (nick) [inline dapps/marketplace/src/pages/listing/_ListingPending.js]: I think we should try to use the `<fbt>` component when possible

File: dapps/marketplace/src/pages/listing/_ListingSold.js [hunk 2]
```diff
@@ -2,21 +2,65 @@ import React from 'react'
...
 )
 
 export default Sold
+
+require('react-styl')(`
+  .listing-buy
+    .status
+      text-align: center
+      padding: 1.25rem
+      background-color: #f3f7f9
```
Comment (nick) [inline dapps/marketplace/src/pages/listing/_ListingSold.js]: Are there variables available for these colors? Might need to go through and look at CSS vars again soon

File: dapps/marketplace/src/pages/listing/listing-types/FractionalHourly.js [hunk 3]
```diff
@@ -1,47 +1,103 @@
...
-          {isDifferentTimeZone && (
-            <div>
-              <fbt desc="listingDetail.timeZoneWarning">
-                NOTE: This is different from your time zone of
-                <fbt:param name="userTimeZone">{userTimeZone}</fbt:param>
-              </fbt>
+      {openCalendar && (
+        <ModalComp
+          title={fbt('Availability', 'Availability')}
+          className={`availability-modal`}
```
Comment (nick) [inline dapps/marketplace/src/pages/listing/listing-types/FractionalHourly.js]: className="availability-modal"

File: dapps/marketplace/src/pages/listing/_ListingSold.js (hunk 2 above)
Comment (shahthepro) [reply dapps/marketplace/src/pages/listing/_ListingSold.js]: It was not there. Can add it if you are fine with it.

Comment (nick) [reply dapps/marketplace/src/pages/listing/_ListingSold.js]: Perhaps we should review all the CSS vars to ensure consistency. Lets come back to it later

File: dapps/marketplace/src/pages/listing/_ListingPending.js (hunk 1 above)
Comment (shahthepro) [reply dapps/marketplace/src/pages/listing/_ListingPending.js]: Have changed it. Just want to know if there is any reason to prefer `<fbt>` over `fbt()`?

Comment (nick) [reply dapps/marketplace/src/pages/listing/_ListingPending.js]: I think it's easier to read when the `desc=` param is explicit

**Pull request merged**